### PR TITLE
Retry p2p server if failed to start

### PR DESCRIPTION
**Pull request by BeaudanBrown**
Catch errors when starting local server and retry every 30 seconds
Completes #224 
Completes #211

### CODE REVIEW

File: libtextsecure/message_receiver.js [hunk 1]
```diff
@@ -120,6 +114,28 @@ MessageReceiver.prototype.extend({
...
+      window.log.warn(
+        'Failed to get my loki address to bind server to, will retry in 30 seconds'
+      );
+      setTimeout(this.startLocalServer.bind(this), 30 * 1000);
+    }
+    try {
+      myServerPort = await localLokiServer.start(localServerPort, myLokiIp);
+    } catch (e) {
+      window.log.warn('Failed to start local loki server, will retry in 30 seconds');
+      setTimeout(this.startLocalServer.bind(this), 30 * 1000);
```
Comment (sachaaaaa) [inline libtextsecure/message_receiver.js]: You should bail after setting those timeouts :)

Also small suggestion, let's make the code flow more simple by moving the success or failure action to perform at the end:
Add a `isListening()` method in our `LocalLokiServer` class, that returns `this.server.listening`.
Then back here, basically in each `catch(e)` just prints the appropriate warning.
And only try to start the localLokiServer if we got a LokiIP first.
Then handle the success or failure.
Pseudo code:
```js
try {
  ... // get loki ip
} catch(e) {
  .. // print error
}
if (myLokiP) {
  try {
    .. // start server
  } catch(e) {
    .. // print error
  }
}
if (localLokiServer.isListening()) {
  ... // everything ok
} else {
 setTimeout(...); // retry later
}
```

Comment (sachaaaaa) [reply libtextsecure/message_receiver.js]: Actually I would prefer to ensure that different exception classes are thrown between `getMyLokiIp` and `LocalLokiServer.start`.
That way you could merge the two try/catches and print the correct error message by checking the error class.
```js
try {
  ... // get loki ip
  ... // start server
} catch(e) {
  .. // print error in function of e's class
}

if (localLokiServer.isListening()) {
  ... // everything ok
} else {
 setTimeout(...); // retry later
}
```

Comment (BeaudanBrown) [reply libtextsecure/message_receiver.js]: How about this?
```js
try {
  myLokiIp = await window.lokiSnodeAPI.getMyLokiIp();
  myServerPort = await localLokiServer.start(localServerPort, myLokiIp);
  window.log.info(`Local Server started at ${myLokiIp}:${myServerPort}`);
  libloki.api.broadcastOnlineStatus();
  localLokiServer.on('message', this.handleP2pMessage.bind(this));
} catch (e) {
  if (e instanceof textsecure.LokiIpError) {
    window.log.warn(
      'Failed to get my loki address to bind server to, will retry in 30 seconds'
    );
  } else {
    window.log.warn('Failed to start local loki server, will retry in 30 seconds');
  }
  setTimeout(this.startLocalServer.bind(this), 30 * 1000);
}
```

Comment (sachaaaaa) [reply libtextsecure/message_receiver.js]: Yup! Even better.

File: libtextsecure/message_receiver.js [hunk 2]
```diff
@@ -120,6 +114,28 @@ MessageReceiver.prototype.extend({
     //   all cached envelopes are processed.
     this.incoming = [this.pending];
   },
+  async startLocalServer() {
+    let myLokiIp;
+    let myServerPort;
```
Comment (sachaaaaa) [inline libtextsecure/message_receiver.js]: No need to declare those two outside the `try` block

Comment (BeaudanBrown) [reply libtextsecure/message_receiver.js]: Oh try whoops

**Pull request merged**